client/state/jetpack-connect/selectors.js

```diff
@@ -10,4 +10,12 @@ const isCalypsoStartedConnection = function( state, siteSlug ) {
 	return false;
 };
 
-export default { isCalypsoStartedConnection };
+const getFlowType = function( state, site ) {
+	const siteSlug = site.slug.replace( /.*?:\/\//g, '' );
```

Comment:
What's the replace for here? Also, we should consider leveraging other existing selectors for this, like [`getSiteSlug`](https://github.com/Automattic/wp-calypso/blob/ab552d23f64c7ce447dc640e1312c4983a8d4bec/client/state/sites/selectors.js#L179-L197), which has the added advantage of allowing us to simplify the argument signature of this function to simply accepting a `siteId`.

Last, I suspect issues in #8215 are related to this line. Called from [this line](https://github.com/Automattic/wp-calypso/blob/ab552d23f64c7ce447dc640e1312c4983a8d4bec/client/signup/jetpack-connect/plans.jsx#L186), we shouldn't assume `selectedSite` is always an object (e.g. fresh localStorage), when the `getSelectedSite` function will return `false`.

cc @roccotripaldi @tyxla